### ISSUE TITLE
Why is the docker-qbittorrentvpn repo archived?

### ISSUE THREAD

**Issue by HerbCSO** [DyonR/docker-Jackettvpn#77]
Hi, I know this is off-topic, but I didn't know how else to contact you about this - my apologies for this!

I've been using and enjoying your docker-qbittorrentvpn project (thanks very much for creating that!) and happened to notice that you archived it - but I cannot find a reason why. I totally understand you may no longer have the inclination or time to work on it, but I just wanted to ask if there is a successor project to this, or what the situation is on this.

Again, my apologies for bugging you like this, I just didn't know what else to try.

**Comment by DyonR** [comment]
I never put out a statement or anything about it. I was, and still am, running **[Unraid](https://unraid.net/)**, which is kind of important for context.

I want containers that are simple to understand and always up to date. If possible, they should be compiled from source, and shouldn't have specific code that works for just one VPN provider. It should either work, or you're out of luck. I wouldn't make exceptions to be compatible with just a handful of providers because they did something different. All my containers were created with Unraid in mind, but since containers are cross-platform, they also worked on, for example, Synology systems. I never intented them to be used outside of Unraid.

It all started with this Docker container - [DyonR/docker-Jackettvpn](https://github.com/DyonR/docker-Jackettvpn). I created this one because there was no Jackett container that had a "VPN layer" built-in and which was available in the 'Apps' section in Unraid. I don't even think one exists nowadays. I live in the Netherlands, and pages like The Piratebay and RarBG (rip) were getting blocked by ISPs here due to lawsuits. RarBG was very popular, but when using your own network connection, you couldn't get any results, hence the need for a VPN container.

Because I had no experience with making Docker containers, I forked an already existing project with a "VPN layer", which was [MarkusMcNugen/docker-qBittorrentvpn](https://github.com/MarkusMcNugen/docker-qBittorrentvpn). Since I had Linux knowledge, I was able to replace qBittorrent with Jackett and implement WireGuard; that's how it all started.

The reason I initially created dyonr/docker-qbittorrentvpn was that, at that time, WireGuard was quickly gaining more popularity. The existing containers, like [binhex/arch-qbittorrentvpn](https://github.com/binhex/arch-qbittorrentvpn), didn't have WireGuard support back then. I honestly kind of wanted to compete with Binhex. This time I replaced Jackett from this project (back) with qBittorrent, but with [libtorrent](https://github.com/arvidn/libtorrent) and qBittorrent compiled from source, to always have the most up-to-date version. Due to the ease of switching one program for another, I made a few more 'VPN' dockers.

All was good. Everything was working fine. I updated the containers, maintained them, they had many downloads, I used them, I recommended them to friends, etc. No real reason to archive or stop maintaining, right?

Well, maybe. Non-Unraid users kept reporting issues, and a lot of people with literally no knowledge of Docker kept requesting support. They often had simple-to-solve problems like not running it in privileged mode on their non-Unraid system, even though it was necessary. The number of issues opened kind of stressed me out. I wanted to help people, but many didn't provide proper logs or context for their issues. Many, or nearly all, I couldn't replicate, or they were issues related to the product itself and not the functionality of the container. Some issues were also related to DNS somehow being blocked in the container, resulting in no connection. I never had those problems and didn't know where to start looking. I just got stressed with all the issues coming into my inbox.

Binhex also added support for his containers with WireGuard, so I lost the motivation to keep working on mine. Additionally, I became busier with personal life and lost interest in maintaining Docker containers. I also wasn't very familiar with how the whole "iptables killswitch" worked — but it works (I guess?). It's still used by this container. Querying a torrent website is less of an issue than leaking your IP while torrenting, so I don't care that much. I felt like many issues might have been caused by it, but I had trouble understanding it well.   
  
Jackettvpn will continue to exist, but I won't provide much support for it.  

  ----

**TLDR** (ChatGPT): I've been using [Unraid](https://unraid.net/) and created containers to be simple, but my intention was never to make them cross-platform. My starting point was [DyonR/docker-Jackettvpn](https://github.com/DyonR/docker-Jackettvpn) because there wasn't a Jackett container with a VPN layer, which was required as sites like The Piratebay and RarBG were blocked in the Netherlands.  
  
I used the foundation from [MarkusMcNugen/docker-qBittorrentvpn](https://github.com/MarkusMcNugen/docker-qBittorrentvpn) and integrated it with WireGuard, given its rising popularity, and of course swapped qBittorrent with Jackett. Meanwhile, containers like [binhex/arch-qbittorrentvpn](https://github.com/binhex/arch-qbittorrentvpn) lacked WireGuard support. That's what motivated me to create a separate container, with WireGuard and qBittorrent.

However, the surge of user issues, especially from non-Unraid users, stressed me out. Many couldn't provide clear problem descriptions. The competition, Binhex, upgraded to WireGuard too, which diminished my motivation.

**Comment by DyonR** [comment]
Oh, if it wasn't clear from the post, just use [binhex/arch-qbittorrentvpn](https://github.com/binhex/arch-qbittorrentvpn)  
Edit: I never used Binhex' one, so I have no idea how much different it is from mine.

**Comment by HerbCSO** [comment]
Awesome, thank you so much for the explanation and for the tip. Exactly what I was looking for. Sorry it became such a hassle for you - I hope this is something you can point to in case somebody else comes to you with the same question. ;]

Best of luck and thanks for all the work you put into this!
